packages/itwinui-react/src/core/TransferList/TransferList.tsx

```diff
@@ -0,0 +1,276 @@
+
+  const onKeyDown = (event: React.KeyboardEvent<HTMLLIElement>) => {
+    if (event.altKey) {
+      return;
+    }
+
+    if (
+      event.key === 'Enter' ||
+      event.key === ' ' ||
+      event.key === 'Spacebar'
+    ) {
+      !disabled && onClickEvents();
+      event.preventDefault();
+    }
+  };
+
+  return (
+    <ListItem
+      ref={ref}
+      onClick={onClickEvents}
+      onKeyDown={onKeyDown}
+      active={active}
+      actionable={actionable}
+      tabIndex={disabled ? undefined : -1}
+      {...rest}
+    >
+      {children}
+    </ListItem>
+  );
+}) as PolymorphicForwardRefComponent<'li', TransferListListItemOwnProps>;
+
+// ----------------------------------------------------------------------------
+// TransferList.Label component
+
+type TransferListLabelOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+export type TransferListLabelProps<T extends React.ElementType = 'div'> =
+  PolymorphicComponentProps<T, TransferListLabelOwnProps>;
+
+const TransferListLabel = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', children, className, ...rest } = props;
+
+  return (
+    <Element
+      className={cx('iui-transfer-list-label', className)}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'div', TransferListLabelOwnProps>;
+
+// ----------------------------------------------------------------------------
+// TransferList.Toolbar component
+
+type TransferListToolbarOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+export type TransferListToolbarProps<T extends React.ElementType = 'div'> =
+  PolymorphicComponentProps<T, TransferListToolbarOwnProps>;
+
+const TransferListToolbar = React.forwardRef((props, ref) => {
+  const { as: Element = 'div', children, className, ...rest } = props;
+
+  return (
+    <Element
+      className={cx('iui-transfer-list-toolbar', className)}
+      role={'toolbar'}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'div', TransferListToolbarOwnProps>;
+
+export type TransferListProps = {
+  /**
+   * Content in the transfer list.
+   */
+  children: React.ReactNode;
+} & Omit<CommonProps, 'title'>;
+
+/**
+ * The TransferList component is used to display a list within a box
+ * @example
+ * <TransferList>
+ *   <TransferList.Area>
+ *     <TransferList.List>
+ *       <TransferList.ListItem>Item 1</TransferList.ListItem>
+ *       <TransferList.ListItem>Item 2</TransferList.ListItem>
+ *       <TransferList.ListItem>Item 3</TransferList.ListItem>
+ *       <TransferList.ListItem>Item 4</TransferList.ListItem>
+ *       <TransferList.ListItem>Item 5</TransferList.ListItem>
+ *       <TransferList.ListItem>Item 6</TransferList.ListItem>
+ *     </TransferList.List>
+ *   </TransferList.Area>
+ * </TransferList>
+ */
+export const TransferList = Object.assign(
+  React.forwardRef(
```

Comment:
move this component to a local variable, preferably near the top

also make it polymorphic

Reply:
Fixed